tmt/steps/provision/__init__.py

```diff
@@ -2531,7 +2531,13 @@ def _run_action_phases(phases: list[Action]) -> tuple[list[ActionTask], list[Act
         # the order or their `order` key. We will group provisioning phases
         # not interrupted by action into batches, and run the sequence of
         # provisioning phases in parallel.
-        all_phases = self.phases(classes=(Action, ProvisionPlugin))
+        all_phases = [
+            p for p in self.phases(
+                classes=(
+                    Action,
+                    ProvisionPlugin)) if isinstance(
+                p,
+                Action) or p.enabled_by_when]
```

Comment:
The formatting is really hurting my eyes, bits that belong together are spread across multiple lines :/ How about the following? It follow the recommended "variable, loop, condition" schema for more verbose loop comprehensions that may help from time to time:

```python
all_phases = [
    p
    for p in self.phases(classes=(Action, ProvisionPlugin))
    if isinstance(p, Action) or p.enabled_by_when
]
```

Reply:
fixed in 3d099121